.github/workflows/production-suite.yml

```diff
@@ -181,3 +181,18 @@ jobs:
       - name: Verify install after upgrading postgresql
         if: ${{ matrix.is_bionic }}
         run: sudo mispipe "/tmp/production-verify 2>&1" ts
+
+      - name: Report status
+        env:
+          ZULIP_BOT_KEY: ${{ secrets.ZULIP_BOT_KEY }}
+        if: |
+          ${{ failure() }} &&
+          github.repository == 'zulip/zulip'
+        run: |
+          if [[ "$ZULIP_BOT_KEY" != "" ]]; then
+            BRANCH="$(python3 -c 'import sys; print(sys.argv[1].split("/")[-1])' "$GITHUB_REF")"
+            URI_ESCAPED_TOPIC="$(python3 -c 'import sys; import urllib.parse; print(urllib.parse.quote(sys.argv[1]))' "$BRANCH failing")"
+            curl  -H "Content-Type: application/json" \
+            -X POST -i 'https://chat.zulip.org/api/v1/external/circleci?api_key='"$ZULIP_BOT_KEY"'&stream=automated%20testing&topic='"$URI_ESCAPED_TOPIC" \
+            -d '{"payload": { "branch": "'"$BRANCH"'", "reponame": "'"$GITHUB_REPOSITORY"'", "status": "failed", "build_url": "'"$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID"'", "username": "'"$GITHUB_ACTOR"'"}}'
+          fi
```

Comment:
Is there a way we can avoid copying this block 3 times?  If nothing else, we can put most of it in a little shell script that we call from 3 places.

Reply:
I was thinking the same but only found complicated ways to doing this. Shell script idea sounds better.